isacc_messaging/api/isacc_record_creator.py

```diff
@@ -179,24 +179,31 @@ def get_patient(self, patient_id):
         if result is not None:
             return Patient(result)
 
-    def get_general_practitioner_emails(self, pt: Patient) -> list:
+    def get_practitioners_emails(self, patient_id) -> list:
         emails = []
-        if pt and pt.generalPractitioner:
-            for gp_ref in pt.generalPractitioner:
-                # format of gp_ref.reference: "Practitioner/2"
-                resource_type, resource_id = gp_ref.reference.split('/')
-                result = HAPI_request('GET', resource_type, resource_id)
-                if result is not None:
-                    if resource_type != 'Practitioner':
-                        raise ValueError(f"expected Practitioner in {gp_ref.reference}")
-                    gp = Practitioner(result)
-                    for t in gp.telecom:
-                        if t.system == 'email':
-                            emails.append(t.value)
+        care_plan = self.get_careplan(patient_id)
+        if care_plan and care_plan.careTeam and len(care_plan.careTeam) > 0:
+            # get the referenced CareTeam resource from the care plan
+            # please see https://www.pivotaltracker.com/story/show/185407795
+            # carePlan.careTeam now includes those that follow the patient
+            resource_type, resource_id = care_plan.careTeam[0].reference.split('/')
```

Comment:
if `len(care_plan.careTeam) > 1` we'll silently ignore them.  perhaps this is okay?

Reply:
thanks @pbugni  this will be logged now if happens.